modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -202,3 +202,59 @@ START_TEST(PIDL)
         skip("?\n");
     ILFree(pidl);
 }
+
+START_TEST(ILIsEqual)
+{
+    LPITEMIDLIST p1, p2, pidl;
+
+    p1 = p2 = NULL;
+    ok_int(ILIsEqual(p1, p2), TRUE);
+
+    ITEMIDLIST emptyitem = {};
+    p1 = p2 = &emptyitem;
+    ok_int(ILIsEqual(p1, p2), TRUE);
+
+    ok_int(ILIsEqual(NULL, &emptyitem), FALSE); // These two are not equal for some reason
```

Comment:
I don't know whether the reason could be: NULL is for invalid, and (non-NULL pointer to) empty item is for that.

Reply:
Both NULL and empty normally means desktop/root but I suppose they are technically not equal if we are talking about the pidl content. This is what Windows does and so must we.